tests/unit/testplan/runnable/interactive/test_api.py

```diff
@@ -397,7 +424,38 @@ def test_put(self, api_env):
         compare_json(json_rsp, suite_json, ignored_keys=["runtime_status"])
 
         ihandler.run_test_suite.assert_called_once_with(
-            "MTest1", "MT1Suite1", await_results=False
+            "MTest1", "MT1Suite1", shallow_report=None, await_results=False
+        )
+
+    def test_put_filtered(self, api_env):
+        """Test updating the SingleSuite resource via PUT."""
+        client, ihandler = api_env
+
+        suite_json = ihandler.report["MTest1"]["MT1Suite1"].serialize()
+        suite_json["runtime_status"] = report.RuntimeStatus.RUNNING
+        rsp = client.put(
+            "/api/v1/interactive/report/tests/MTest1/suites/MT1Suite1",
+            json=suite_json,
+        )
+        assert rsp.status_code == 200
+        json_rsp = rsp.get_json()
+        assert json_rsp["runtime_status"] == report.RuntimeStatus.WAITING
```

Comment:
Same as above. I can remove an entry from the JSON sent to the put as that would emulate what happens functionally with a real UI. However, the response is shallow and I cannot work with the entries, I can only ensure that the correct way of the logic is triggered. While I am writing this, one idea: make this test just on asserting the correct call, but add tests for the irunner itself which operates the actual report so I can see the filtered run. For runtime status I really do not know how to test entry level. I can unit test the alternative setter. So basically instead of this single test_api item, I would have three unit tests for each type of entry. How does it sound?